services/QuillLMS/spec/models/activity_spec.rb

```diff
@@ -164,6 +164,27 @@
       expect(activity.module_url(student.activity_sessions.build()).to_s).to include "student"
     end
 
+    it "must add 'activities' param if it the student has completed previous sessions of this activity classification" do
+      classification = create(:activity_classification, key: 'connect')
+      classified_activity = create(:activity, classification: classification)
+      student = create(:student)
+      uac = create(:user_activity_classification, user: student, activity_classification: classification, count: 4)
+      activity_session = create(:activity_session, activity: classified_activity, user: student)
+
+      classification.reload
+
+      expect(activity_session.classification.user_activity_classifications.where(id: student.id).first.count).to eq(4)
```

Comment:
Some suggestions for refactoring this expectation:

1.  Change finder attribute from `id` to `user_id`
2.  Add an association `has_many :user_activity_classifications, through: :classification` to activity_session.rb 
3.  Use `find_by` instead of [where(...).first](https://github.com/rubocop/rails-style-guide#find_by)
4.  Since the expectation is pulling a record, check equality with the record rather than the attribute.

 `expect(activity_session.user_activity_classifications.find_by(user_id: student.id)).to eq(uac)`